dependency charts upgrade and misc upgrade improvement/fix

Vicente-Cheng opened the pull request:
related issue: #3159, #2115

Bump several needed charts and misc fixes

Vicente-Cheng committed:
chart: bump several charts to the latest

    - bump pci-device/vm-import for the upgrade path
    - bump harvester-network-controller to v0.3.1
    - bump harvester-node-disk-manager to v0.4.8
    - bump harvester-node-manager to v0.1.3
    - change the harvester-node-manager tag to master-head on master
      branch

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

guangbochen reviewed (commented):
also need to patch the upgrade version check on https://github.com/harvester/harvester/blob/master/package/upgrade/upgrade_manifests.sh#L619, thanks.

guangbochen reviewed (approved):
LGTM, thanks.

starbops reviewed (approved):
Tested on v1.0.3 upgrade. LGTM, thanks!

ibrokethecloud reviewed (approved):
LGTM

starbops reviewed (approved):
Tested upgrade from v1.0.3 again and LGTM, thanks!

Vicente-Cheng committed:
upgrade: improve the upgrade behavior for addons

    - We should not only change the `version`.
      The `tag` also need to handle for the right version.

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

Vicente-Cheng committed:
upgrade: add default grubenv to prevent search hang

    - we also need to apply this change on the upgrade path

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

ibrokethecloud reviewed (approved):
LGTM.

Tried out an upgrade, and can see the addons are patched, with both `version` and `valuesContent` getting updated.

```
Logging Event Audit: nothing to do in v1.1.0
Applying extra manifests
Warning: resource settings/default-data-path is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
setting.longhorn.io/default-data-path configured
Warning: resource customresourcedefinitions/ippools.whereabouts.cni.cncf.io is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
customresourcedefinition.apiextensions.k8s.io/ippools.whereabouts.cni.cncf.io configured
Warning: resource customresourcedefinitions/overlappingrangeipreservations.whereabouts.cni.cncf.io is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
customresourcedefinition.apiextensions.k8s.io/overlappingrangeipreservations.whereabouts.cni.cncf.io configured
addon.harvesterhci.io/vm-import-controller patched
addon.harvesterhci.io/pcidevices-controller patched

demo:~ # kubectl get addons.harvesterhci -o yaml -n harvester-system
apiVersion: v1
items:
- apiVersion: harvesterhci.io/v1beta1
  kind: Addon
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"harvesterhci.io/v1beta1","kind":"Addon","metadata":{"annotations":{},"labels":{"addon.harvesterhci.io/experimental":"true"},"name":"pcidevices-controller","namespace":"harvester-system"},"spec":{"chart":"harvester-pcidevices-controller","enabled":false,"repo":"http://harvester-cluster-repo.cattle-system.svc/charts","valuesContent":"image:\n  tag: v0.2.2\nfullnameOverride: harvester-pcidevices-controller\n","version":"0.2.2"}}
    creationTimestamp: "2022-11-25T05:22:47Z"
    generation: 2
    labels:
      addon.harvesterhci.io/experimental: "true"
    name: pcidevices-controller
    namespace: harvester-system
    resourceVersion: "85398"
    uid: 59c1c0c5-acd2-45e7-a2f1-77e30f0a9305
  spec:
    chart: harvester-pcidevices-controller
    enabled: false
    repo: http://harvester-cluster-repo.cattle-system.svc/charts
    valuesContent: |
      image:
        tag: v0.2.3
      fullnameOverride: harvester-pcidevices-controller
    version: 0.2.3
- apiVersion: harvesterhci.io/v1beta1
  kind: Addon
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"harvesterhci.io/v1beta1","kind":"Addon","metadata":{"annotations":{},"name":"vm-import-controller","namespace":"harvester-system"},"spec":{"chart":"harvester-vm-import-controller","enabled":false,"repo":"http://harvester-cluster-repo.cattle-system.svc/charts","valuesContent":"image:\n  tag: v0.1.1\nfullnameOverride: harvester-vm-import-controller\n","version":"0.1.1"}}
    creationTimestamp: "2022-11-25T05:22:47Z"
    generation: 2
    name: vm-import-controller
    namespace: harvester-system
    resourceVersion: "85390"
    uid: 5cd68687-1896-4ea1-9eed-fac455b2a168
  spec:
    chart: harvester-vm-import-controller
    enabled: false
    repo: http://harvester-cluster-repo.cattle-system.svc/charts
    valuesContent: |-
      image:
        tag: v0.1.2
      fullnameOverride: harvester-vm-import-controller
    version: 0.1.2
kind: List
metadata:
  resourceVersion: ""
```

bk201 reviewed (approved):
LGTM

guangbochen reviewed (approved):
LGTM and verified that the upgrade works, thanks.

starbops reviewed (approved):
LGTM, thanks!
